Zend/zend_builtin_functions.c

```diff
@@ -633,13 +633,12 @@ ZEND_FUNCTION(define)
 	ZEND_PARSE_PARAMETERS_END();
 
 	if (zend_memnstr(ZSTR_VAL(name), "::", sizeof("::") - 1, ZSTR_VAL(name) + ZSTR_LEN(name))) {
-		zend_error(E_WARNING, "Class constants cannot be defined or redefined");
+		zend_value_error("Class constants cannot be defined or redefined");
 		RETURN_FALSE;
 	}
 
 	if (non_cs) {
-		zend_error(E_WARNING,
-			"define(): Declaration of case-insensitive constants is no longer supported");
+		zend_value_error("define(): Declaration of case-insensitive constants is no longer supported");
```

Comment:
should be `zend_argument_value_error(3, "...")`, but we could just remove the param if we throw an exception anyway. (I am not sure if we really can)

Reply:
Throwing away the argument seems to be the best thing to do. But that's an independent change.